parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs

```diff
@@ -643,12 +643,18 @@ ensureNameLookupForBranchHash getDeclType mayFromBranchHash toBranchHash = do
       toBranch <- Ops.expectBranchByBranchHash toBranchHash
       treeDiff <- BranchDiff.diffBranches fromBranch toBranch
       let namePrefix = Nothing
-      let BranchDiff.NameChanges {termNameAdds, termNameRemovals, typeNameAdds, typeNameRemovals} = BranchDiff.nameChanges namePrefix treeDiff
-      termNameAddsWithCT <- do
-        for termNameAdds \(name, ref) -> do
-          refWithCT <- addReferentCT ref
-          pure $ toNamedRef (name, refWithCT)
-      Ops.buildNameLookupForBranchHash mayExistingLookupBH toBranchHash (termNameAddsWithCT, toNamedRef <$> termNameRemovals) (toNamedRef <$> typeNameAdds, toNamedRef <$> typeNameRemovals)
+      -- TODO: Might be worth switching to a streaming lib to make this less confusing.
+      Ops.buildNameLookupForBranchHash
+        mayExistingLookupBH
+        toBranchHash
+        ( \save -> do
```

Comment:
The nested callback stuff is a bit annoying;

Totally open to other options, e.g. using a TBQueue or something, but note that those are also slightly annoying because we'd then want to fork a separate worker, and doing unlifting and IO is annoying in its own right since we don't want to add a MonadUnliftIO instance for Transaction.

TLDR; this is a little mindbending but seemed like the safest and lesser of the evils, it allows us to stay in Transaction the whole time and means we'll keep at most one namespace diff in memory at a time.